File: backend/app/models/task.py
import enum
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy import ForeignKey
from .base import ModelBase
from datetime import datetime

class Status(enum.Enum):
    pending = "pending"
    not_started = "not started"
    in_progress = "in progress"
    completed = "completed"

class Task(ModelBase):
    __tablename__ = "task"
    
    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    title: Mapped[str] = mapped_column(index=True)
    description: Mapped[str]
    creation_date: Mapped[datetime] = mapped_column(default=datetime.now())
    finish_date: Mapped[datetime]
    status: Mapped[Status] = mapped_column(default=Status.pending)
    user_id: Mapped[int] = mapped_column(ForeignKey("user.id"))
    category_id: Mapped[int] = mapped_column(ForeignKey("category.id"))
 
    
    

File: frontend/src/types/task.ts
interface Task {
  id: number;
  title: string;
  description: string;
  creation_date: string;
  finish_date: string;
  status: string;
  category_id: number;
  user_id: number;
}

interface TaskData {
  title: string;
  description: string;
  category_id: number | string;
  status: string;
  finish_date: Date | undefined;
}

interface UpdateTask {
  title: string;
  description: string;
  status: string;
}

export type { Task, TaskData, UpdateTask };
